examples/timelock-controller/src/contract.rs

```diff
@@ -0,0 +1,347 @@
+//! Timelock Controller Example Contract.
+//!
+//! This contract demonstrates a complete timelock controller implementation
+//! with role-based access control, similar to OpenZeppelin's Solidity
+//! TimelockController.
+//!
+//! # Roles
+//!
+//! - **Admin**: Can manage all roles and update the minimum delay. By default,
+//!   the contract itself is the admin, meaning admin operations must go through
+//!   the timelock process.
+//! - **Proposer**: Can schedule operations. Proposers are also automatically
+//!   granted the Canceller role.
+//! - **Executor**: Can execute operations that are ready.
+//! - **Canceller**: Can cancel pending operations.
+//!
+//! # Usage
+//!
+//! ## Standard Operations (External Contracts)
+//!
+//! 1. Deploy the contract with initial proposers, executors, and minimum delay
+//! 2. Proposers schedule operations with a delay >= minimum delay
+//! 3. After the delay passes, executors call `execute_op` to execute operations
+//! 4. Cancellers can cancel operations before they are executed
+//!
+//! ## Self-Administration
+//!
+//! When the contract is deployed with `admin` set to `None`, the contract
+//! address itself becomes the admin, enabling self-administration. For
+//! self-administration operations (e.g., updating the minimum delay):
+//!
+//! 1. Proposer schedules the operation targeting the timelock contract itself
+//! 2. After the delay passes, call the admin function directly (not via
+//!    `execute_op`)
+//! 3. The `CustomAccountInterface` implementation validates the operation is
+//!    ready and marks it as executed
```

Comment:
I'm not familiar with the Solidity one, and this is the first time I'm reading about the subject. And I'm still confused about this `Self-Administration`.

> //! 2. After the delay passes, call the admin function directly (not via `execute_op`)

Who calls? Does the caller need specific privileges, like `admin`? If so, aren't they become the admin in this scenario? Why is it called `Self-Administration`? What does it exactly mean the contract itself become the admin? Then I expect no-one to call the admin functions, but administration functions will be called maybe indirectly throughout other operations with some triggers. 

The explanations here puzzled me...